quix-frontend/client/src/lib/sql-autocomplete/adapter/sql-autocomplete-adapter-utills.ts

```diff
@@ -0,0 +1,193 @@
+  brokenPrefix = brokenPrefix.filter(cell => cell.trim() !== '');
+
+  const relevantPrefixes: string[] = [];
+
+  for (const table of tables) {
+    for (const column of table.columns) {
+      let found = false;
+      let gotRelevantPartOfPrefix = false;
+      let currentPrefix = '';
+
+      brokenPrefix.forEach(cell => {
+        if (found) {
+          currentPrefix += cell + '.';
+        }
+        if (typeof column === 'object' && !gotRelevantPartOfPrefix && cell === column.name) {
+          found = true;
+          gotRelevantPartOfPrefix = true;
+          currentPrefix += cell + '.';
+        }
+      });
+
+      if (currentPrefix) {
+        relevantPrefixes.push(currentPrefix);
+      }
+    }
+  }
+
+  if (relevantPrefixes.length === 0) {
+    return ''; // No relevant prefixes found
+  }
+
+  // Find the longest relevant prefix
+  let longestPrefix = relevantPrefixes[0];
+  for (const prefix of relevantPrefixes) {
+    if (prefix.length > longestPrefix.length) {
+      longestPrefix = prefix;
+    }
+  }
+
+  return longestPrefix
+}
+
+
+
+function getAllChildrenOfTables(tables: TableInfo[]): any {
+  let allChildren = []
+  for (const extractedTable of tables) {
+    const { columns } = extractedTable;
+      columns.forEach((column) => {
+        if ( typeof column === 'object' && typeof column.dataType === "string") {
+          column.dataType = trinoToJs(column.dataType, 0);
+        }
+        if (typeof column === 'object' && typeof column.dataType === "object")  {
+          allChildren = allChildren.concat(getObjectChildren(column.dataType, column.name));
+        }
+  });
+}
+return allChildren;
+}
+
+export function getSearchCompletion(tables: TableInfo[] , prefix: string | undefined):any {
+  if(!prefix.includes('.')) {
+    return [];
+  }
+  const allChildren = getAllChildrenOfTables(tables);
+  const relevantPartOfPrefix = findRelevantPartOfPrefix(tables , prefix.split('.')).slice(0, -1);
+  // const relevantPartOfPrefix = findRelevantPartOfPrefix(tables , prefix.split('.')).slice(0, -1);
+  if( !relevantPartOfPrefix )  {
+    return []
+  }
+  const lastDotIndex = relevantPartOfPrefix.lastIndexOf('.');
+  const startOfSearch = lastDotIndex !== -1 ? relevantPartOfPrefix.slice(0, lastDotIndex + 1) : relevantPartOfPrefix;
+  const searchPart = relevantPartOfPrefix.replace(startOfSearch,'')
+  const filteredChildren = allChildren.filter(obj => {
+  const nameInLowerCase = obj.name.toLowerCase();
+  const parts = nameInLowerCase.split('.');
+    if (parts.length > 1) {
+        const substringAfterFirstDot = parts.slice(1).join('.');
+        const criteria = checkCriteria(substringAfterFirstDot , searchPart.toLowerCase());
+        const flterIfInDiffrenCollumn = obj.name.startsWith(relevantPartOfPrefix.split('.')[0]);
+        return obj.name.includes(startOfSearch) && substringAfterFirstDot.includes(searchPart.toLowerCase()) && criteria && flterIfInDiffrenCollumn ;
+    }
```

Comment:
```suggestion
        const flterIfInDiffrenColumn = obj.name.startsWith(relevantPartOfPrefix.split('.')[0]);
        return obj.name.includes(startOfSearch) && substringAfterFirstDot.includes(searchPart.toLowerCase()) && criteria && flterIfInDiffrenColumn ;
    }
```

Reply:
Pleaseeeeeeeeee 
download Code Spell Checker plugin for vs code!

https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

use it! it's necessary!

Reply:
im sorry  😢